### ISSUE TITLE
Display CJK characters correctly in non-fallback languages

### ISSUE THREAD

**Issue by zaoqi** [minetest/minetest#9198]
## Problem

cannot display CJK characters

## Solutions

use another font?

## Additional context

(partially) fix https://github.com/minetest/minetest/issues/4638

**Comment by SmallJoker** [comment]
Is Irrlicht capable of reading your inputs? Its input system is broken, thus I wonder whether you could really type non-latin characters.

**Comment by zaoqi** [comment]
I renamed a world to a non-latin name

**Comment by Enrikoo** [comment]
What if someone doesn't like the font and wants a refactor of arimo?

**Comment by zaoqi** [comment]
@Enrikoo it can change font

**Comment by zaoqi** [comment]
@Enrikoo
> What if someone doesn't like the font and wants a refactor of arimo?

If someone doesn't like the default English or CJK font, he can change font files.

**Comment by zaoqi** [comment]
@SmallJoker I found that minetest already has a CJK font, but only enabled when `needs_fallback_font` is yes. I want minetest able to display CJK characters when the language is en.

**Comment by SmallJoker** [comment]
Translator's fault. Will be fixed in next release unless someone reverts the translation change.

**Comment by zaoqi** [comment]
@SmallJoker 
The problem is that minetest cannot display CJK characters when language is en (`needs_fallback_font` is not `yes`)
should I open an new issue?

**Comment by v-rob** [comment]
This shouldn't have a "Bug" label.  Many fonts (Minetest's normal and monospace fonts included) don't support CJK characters.  It really can't be considered a bug since no font has every character in Unicode.  However, one can make it better by doing `font_path = fonts/DroidSansFallbackFull.ttf`.

However, Minetest could try to better handle these problems by doing the following: if an unknown character is encountered, attempt to draw the character with the fallback font.  If that fails, draw a question mark or empty box to indicate that it couldn't be drawn.  However, such a thing is non-trivial since fonts are drawn all over Minetest, such as in the GUI, HUD, nametags, debug text, etc.  In addition, Unicode has the ability to combine characters, so there has to be a way to detect character combinations and render them all with the fallback font together (ugh, Unicode is so complicated, even more so with UTF-8 to worry about).

In other words, this can't reasonably done without reworking the font system completely and making everything use that font system.

**Comment by sfan5** [comment]
I disagree that this is non-trivial. We have a central class (`CGUITTFont`) that already *is* used everywhere, surely it can be made to handle this.

**Comment by v-rob** [comment]
Perhaps you're right, but it can't be done with `CGUITTFont` alone because that can only handle one font at a time, and only Freetype fonts at that.  If there were some sort of grouped font, it could work, but it would still be tricky.
